src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -285,11 +285,11 @@ String classData(Object arg) {
         return name;
     }
 
-    List<Object> classDataValues() {
+    private Object classDataValues() {
         final List<ClassData> cd = classData;
         return switch(cd.size()) {
-            case 0 -> List.of();
-            case 1 -> List.of(cd.get(0).value);
+            case 0 -> null;
```

Comment:
`List.of()` always returns the same singleton instance and does not cause any object allocation.  I prefer to keep the `classDataValues()` to return `List<Object>`

Reply:
Ah, I now see why you have to change the signature because of the single element case.  I made my comment too quickly. 

A couple of suggestions:
Please add a javadoc for this method to describe what this method returns for different cases.   It's better to move this method close to `clinit` as they are tightly coupled.

Reply:
I moved the method and added in javadoc. Could you check if the wording is OK?